src/EditorFeatures/CSharpTest/RemoveUnnecessaryParentheses/RemoveUnnecessaryParenthesesTests.cs

```diff
@@ -0,0 +1,1655 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.RemoveUnnecessaryParentheses;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.RemoveUnnecessaryParentheses
+{
+    public partial class RemoveUnnecessaryParenthesesTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
```

Comment:
❕ Additional test cases:

```
(this.Property) = Property;
```

```
x += (3 * 2);
```

```
x *= (3 + 2);
```

```
((short)3).ToString();
```

```
int x = 3 * (checked(5));
```

```
int x = 3 * (unchecked(5));
```

```
string property = "My " + (nameof(property));
```

```
bool x = ("" is string);
```

```
string x = ("" is string).ToString();
```

```
string y;
string x = (y = "text");
```

```
Func<string, string> y1 = (v) => v;
Func<string, string> y2 = (v => v);
Func<string, string> y3 = ((v) => v);
Func<string, string> z1 = () => "";
Func<string, string> z2 = (() => "");
```

```
string y = ((Func<string, string>)((v) => v))("text");
```

```
(await Method()).OtherMethod();
```

```
#if (A || B)
#if(A)
#if (A)
```

```
return(value);
return (value);
```

Reply:
Note: the code we're trying to format looks like this:

```c#
    {
#ifA || B
#endif
    }
```

There is elastic trivia between the ```if``` and the ```A```.  But this doesn't actually get formatted.  I can see the formatter examining the ```{``` and ```}``` as the only pair of tokens to look at.  But i can't see it ever diving into the directives.  I'm not sure how this is supposed to work?  Can you advise?